content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+Our final goal is to create a package that is "downloadable" by users, which means this package needs a GitHub repository so that it can be easily shared. If students wanted to, they’d be able to find the source code and solutions in your GitHub repository (or, with some spelunking, in the folder that houses all their R packages locally). But this would take some extra effort on the part of students.
+
+If a solution like this still works for your purposes, then read onwards!
+
+
+### Pre-requisites
+
+We assume the following:
+
+* You have a GitHub account (but you don't need to be more than a GitHub novice)
+* You are comfortable with the basics of using R and RStudio
+* You understand how R projects work
+* You are already familiar with creating a regular learnr tutorial
+
+
+
+### Getting set up
+
+Let's make sure we have all the pieces in place to create a learnr tutorial package.
+
+
+#### Packages
+
+Make sure you've installed and loaded the following packages:
+
+```{r, eval = FALSE}
+#install.packages(c("usethis", "devtools", "roxygen2"))
+library(usethis)
+library(roxygen2)
+library(devtools)
+library(learnr)
+```
+
+
+#### Create a GitHub repository
+
+If you're already familiar with the ins and outs of connecting an R Project to a GitHub repository, you can skip this section.
+We recommend following the tips on [Happy Git with R](https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2) and starting in GitHub before switching to RStudio.
+
+1. [**Create a new repository on GitHub**](https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2) for your work.
+
+    * The repository name you choose will also end up being the name of your package, so choose wisely! For the examples in this guide, we'll go with "tutorialpackage".
+
+1. **Copy the repository URL** to your clipboard.
+
+    * Do this by clicking the green Clone or Download button.
+    * Copy the HTTPS URL (looks like: `https://github.com/{yourname}/{yourrepo}.git`).
+
+    * **Or**, copy the SSH URL if you chose to set up SSH keys (looks like: `git@github.com:{yourname}/{yourrepo}.git`).
+
+1. **Open RStudio** and [create a new RStudio Project via git clone](https://happygitwithr.com/new-github-first.html#new-rstudio-project-via-git-clone).
+
+    * Do this by clicking File > New Project > Version Control > Git.
+
+    * Paste the copied URL.
+
+    * Be intentional about where you tell RStudio to create this new Project.
+
+1. **Click Create Project.**
+
+    * Your R Project is now connected to GitHub!
+
+
+
+### Add files and directories for a package
+
+Now we'll create the infrastructure to turn this R project into a *package*. A handful of new files need to be created, but luckily we can outsource this heavy lifting to the `create_package()` function from the [usethis](https://usethis.r-lib.org/) package.
```

Comment:
may be a good spot to link to: https://twitter.com/allison_horst/status/1302354103968780289?s=20

I messaged Allison Horst b/c the illustration is not yet in her GitHub repo, will report back

Reply:
https://github.com/allisonhorst/stats-illustrations/blob/master/rstats-artwork/usethis.png